lua/caramba/multifile.lua

```
@@ -388,55 +388,104 @@ end
-        line = tonumber(lnum),
-        col = tonumber(col),
-        text = text,
-      })
+  local root = opts.directory or vim.fn.getcwd()
+  local cfg = require("caramba.config").get()
+  local scandir = require('plenary.scandir')
+
+  -- 1. Scan for all relevant files in the project
+  local all_files = scandir.scan_dir(root, { hidden = false, respect_gitignore = true })
+  local files_to_check = {}
+  local include_extensions = cfg.search.include_extensions or { 'lua', 'py', 'js', 'ts', 'jsx', 'tsx', 'go', 'rs', 'rb', 'php' }
```

Luacheck (1.2.0)

[warning] line is too long (128 > 120)

(W631)